File: Lib/Telerik/Examples/QuickStart/ChartView/LiveData/ExampleCustomShapeControl.vb
Imports System.Text
Imports Telerik.WinControls
Imports System.ComponentModel

Namespace Telerik.Examples.WinControls.ChartView.LiveData
	<ToolboxItem(True)>
	Friend Class ExampleCustomShapeControl
		Inherits RadControl
'INSTANT VB NOTE: The variable element was renamed since Visual Basic does not allow class members with the same name:
		Private element_Renamed As ExampleCustomShapeElement

		Public ReadOnly Property Element() As ExampleCustomShapeElement
			Get
				Return element_Renamed
			End Get
		End Property

		Public Property LeftText() As String
			Get
				Return Me.element_Renamed.LeftText
			End Get
			Set(ByVal value As String)
				Me.element_Renamed.LeftText = value
			End Set
		End Property

		Public Property RightText() As String
			Get
				Return Me.element_Renamed.RightText
			End Get
			Set(ByVal value As String)
				Me.element_Renamed.RightText = value
			End Set
		End Property

		Public Sub New()

		End Sub

		Protected Overrides ReadOnly Property DefaultSize() As Size
			Get
				Return New Size(200, 30)
			End Get
		End Property

		Protected Overrides Sub CreateChildItems(ByVal parent As RadElement)
			Me.element_Renamed = New ExampleCustomShapeElement()
			Me.RootElement.Children.Add(element_Renamed)
			MyBase.CreateChildItems(parent)
		End Sub
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/ChartView/LiveData/ExampleCustomShapeControl.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls;
using System.Drawing;
using System.ComponentModel;

namespace Telerik.Examples.WinControls.ChartView.LiveData
{
    [ToolboxItem(true)]
    class ExampleCustomShapeControl : RadControl
    {
        private ExampleCustomShapeElement element;

        public ExampleCustomShapeElement Element
        {
            get { return element; }
        }

        public string LeftText 
        {
            get 
            {
                return this.element.LeftText;
            }
            set 
            {
                this.element.LeftText = value;
            }
        }

        public string RightText
        {
            get
            {
                return this.element.RightText;
            }
            set
            {
                this.element.RightText = value;
            }
        }

        public ExampleCustomShapeControl() 
        {

        }

        protected override Size DefaultSize
        {
            get
            {
                return new Size(200, 30);
            }
        }

        protected override void CreateChildItems(RadElement parent)
        {
            this.element = new ExampleCustomShapeElement();
            this.RootElement.Children.Add(element);
            base.CreateChildItems(parent);
        }
    }
}
